pkg/config/output_formats.go

```diff
@@ -0,0 +1,57 @@
+package config
+
+type Formats struct {
+	Text        Text         `mapstructure:"text"`
+	JSON        SimpleFormat `mapstructure:"json"`
+	Tab         Tab          `mapstructure:"tab"`
+	HTML        SimpleFormat `mapstructure:"html"`
+	Checkstyle  SimpleFormat `mapstructure:"checkstyle"`
+	CodeClimate SimpleFormat `mapstructure:"code-climate"`
+	JUnitXML    JUnitXML     `mapstructure:"junit-xml"`
+	TeamCity    SimpleFormat `mapstructure:"teamcity"`
+	Sarif       SimpleFormat `mapstructure:"sarif"`
+}
+
+func (f *Formats) IsEmpty() bool {
+	styles := []SimpleFormat{
+		f.Text.SimpleFormat,
+		f.JSON,
+		f.Tab.SimpleFormat,
+		f.HTML,
+		f.Checkstyle,
+		f.CodeClimate,
+		f.JUnitXML.SimpleFormat,
+		f.TeamCity,
+		f.Sarif,
+	}
+
+	for _, style := range styles {
+		if style.Path != "" {
+			return false
+		}
+	}
+
+	return true
+}
+
+type SimpleFormat struct {
+	Path string `mapstructure:"path"`
+}
```

Comment:
nit: Can be refactored to:
```go
func (f *Formats) IsEmpty() bool {
	styles := []interface {
		IsEmpty() bool
	}{
		&f.Text,
		&f.JSON,
		&f.Tab,
		&f.HTML,
		&f.Checkstyle,
		&f.CodeClimate,
		&f.JUnitXML,
		&f.TeamCity,
		&f.Sarif,
	}

	for _, style := range styles {
		if !style.IsEmpty() {
			return false
		}
	}

	return true
}

type SimpleFormat struct {
	Path string `mapstructure:"path"`
}

func (s *SimpleFormat) IsEmpty() bool {
	return s.Path == ""
}
```

Reply:
This is less readable, and not useful as `SimpleFormat` is already shared between all the types.